### PR TITLE
EIP-6914: Update the equivocating indices in the fork-choice store

### PR DESCRIPTION
**Pull request by ppopth**
Since, in EIP-6914, we sometimes reuse the slashed validator index, we need to remove it from the list of equivocating indices in the fork-choice store.

See https://github.com/ethereum/consensus-specs/issues/3335#issuecomment-1600262322 for discussion. Thank @dapplion for the solution.

### CODE REVIEW

File: specs/_features/eip6914/fork-choice.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Introduction
+
+This is the modification of the fork choice according to EIP-6914.
+
+## Fork choice
+
+A new handler is added with this upgrade:
+
+- `on_reused_index(store, index)` whenever a validator index `index: ValidatorIndex` is reused
```
Comment (djrtwo) [inline specs/_features/eip6914/fork-choice.md:25]: This is a bit wordy, but I'd like to provide a more specific clarification of what this means.

```suggestion
- `on_reused_index(store, index)` whenever a validator index `index: ValidatorIndex` is reused. That is [`get_index_for_new_validator()`](./beacon-chain.md#get_index_for_new_validator) provides an index due a return value of  `True` from [`is_reusable_validator()`](./beacon-chain.md#is_reusable_validator).
```

**Review by djrtwo — APPROVED** [review]
really nice catch!

I want to try to be more specific about when this is called, see below

**Review by dapplion — APPROVED** [review]
Looks good, thank you!

**Review by badasserry — APPROVED** [review]

**Review by badasserry — APPROVED** [review]
